src/lib.rs

```diff
@@ -2,6 +2,34 @@
 
 //! Module level docs
 
+/// Possible errors when parsing BUFR messages
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// Message too short to even contain the total size
+    #[error("BUFR message is too short, less than 8 bytes")]
```

Comment:
We can use https://github.com/yaahc/displaydoc#example to avoid repeating the message and the docs